### PR TITLE
Lattice join

### PR DESCRIPTION
**Pull request by zzlk**

### CODE REVIEW

File: hydroflow_lang/src/graph/ops/lattice_join.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    soft_range_inn: &(2..=2),
+    hard_range_out: RANGE_1,
+    soft_range_out: RANGE_1,
+    num_args: 0,
+    persistence_args: &(0..=2),
+    type_args: &(2..=2),
+    is_external_input: false,
+    ports_inn: Some(|| super::PortListSpec::Fixed(parse_quote! { 0, 1 })),
+    ports_out: None,
+    properties: FlowProperties {
```
Comment (zzlk) [inline hydroflow_lang/src/graph/ops/lattice_join.rs:81]: looking for feedback on how this operator interacts with the flow properties, I'm not sure if it is monotonic/deterministic/etc.

File: hydroflow/src/compiled/pull/symmetric_hash_join_lattice.rs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            }
+            Entry::Vacant(e) => {
+                e.insert(v);
+                true
+            }
+        }
+    }
+}
+
+pub type _JoinStateLattice<Key, V1: LatticeRepr, V1D, V2: LatticeRepr, V2D> = (
```
Comment (zzlk) [inline hydroflow/src/compiled/pull/symmetric_hash_join_lattice.rs:45]: probably just remove this whole type completely if it is unusued.

File: hydroflow/src/compiled/pull/symmetric_hash_join_lattice.rs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+#[cfg(test)]
+mod tests {
+    use super::{SymmetricHashJoinLattice, _JoinStateLattice as JoinStateLattice};
+    use crate::lang::lattice::ord::MaxRepr;
+
+    type JoinStateMaxLattice =
+        JoinStateLattice<usize, MaxRepr<usize>, MaxRepr<usize>, MaxRepr<usize>, MaxRepr<usize>>;
+
+    fn join<X: IntoIterator<Item = (usize, usize)>, Y: IntoIterator<Item = (usize, usize)>>(
```
Comment (zzlk) [inline hydroflow/src/compiled/pull/symmetric_hash_join_lattice.rs:153]: rename X to LHS and Y to RHS

**Review by MingweiSamuel — CHANGES_REQUESTED** [review]
pedantic: acronyms should only have their first letter capitalized, `Lr`, `LrDelta`, `Lhs`, `LhsDelta`, etc

File: hydroflow_lang/src/graph/ops/lattice_join.rs (hunk 1 above)
Comment (MingweiSamuel) [reply hydroflow_lang/src/graph/ops/lattice_join.rs:81]: I think `preserve, preserve, false` is right

**Review by MingweiSamuel — APPROVED** [review]

File: hydroflow_lang/src/graph/ops/lattice_join.rs (hunk 1 above)
Comment (MingweiSamuel) [reply hydroflow_lang/src/graph/ops/lattice_join.rs:81]: We might need a more complex mechanism for multi-input operators. but for now the flowproperties are unused

**Pull request merged**